pkg/scheduler/scheduler.go

```diff
@@ -282,25 +282,31 @@ func (s *Scheduler) cleanUpAllBindingsFor(ctx context.Context, crp *fleetv1beta1
 		return controller.NewAPIServerError(false, err)
 	}
 
-	// Remove the scheduler cleanup finalizer from all the bindings, and delete them.
-	//
+	// Remove scheduler CRB cleanup finalizer from deleting bindings.
+	for idx := range bindingList.Items {
+		binding := &bindingList.Items[idx]
+		controllerutil.RemoveFinalizer(binding, fleetv1beta1.SchedulerCRBCleanupFinalizer)
+		if err := s.client.Update(ctx, binding); err != nil {
+			klog.ErrorS(err, "Failed to remove scheduler reconcile finalizer from cluster resource binding", "clusterResourceBinding", klog.KObj(binding))
+			return controller.NewUpdateIgnoreConflictError(err)
+		}
+	}
```

Comment:
why not combine this loop with the one below?

Reply:
I am not sure if there are tests about this?

Reply:
We have,
UT - https://github.com/Azure/fleet/pull/924/files#diff-130e36b28b6a84d355879caa6b7927c98ec18c63adad6ab4fc486458763080bfR56
IT - Where we delete CRPs and expect all CRBs to be deleted https://github.com/Azure/fleet/blob/main/test/scheduler/pickall_integration_test.go#L60